src/result.rs

```diff
@@ -1,4 +1,4 @@
-use error::Error;
+use core::error::Error;
 
 /// `Result` type for CBOR serialisation and deserialisation.
-pub type Result<T> = ::std::result::Result<T, Error>;
+pub type Result<T> = core::result::Result<T, dyn Error>;
```

Comment:
The problem is that you're trying to use a bare trait object which just isn't possible in Rust, trait objects are unsized types that always have to be behind a pointer type, so you'll need some container type to act as the sized type that can point to the erased error type in the trait object.

I'm assuming this won't work because you're doing no_std and probably can't assume an allocator but the default suggestion to fix this would be to Box it.
```suggestion
pub type Result<T> = core::result::Result<T, Box<dyn Error + Send + Sync + 'static>>;
```

Reply:
Thanks for the suggestion and help. I have to admit, I'm pretty new to rust^^

Reply:
My pleasure, and welcome!